supervisor/supervisor.go

```diff
@@ -275,8 +277,12 @@ func (s *Supervisor) Start() error {
 		"cpus":        s.machine.Cpus,
 	}).Debug("containerd: supervisor running")
 	go func() {
-		for i := range s.tasks {
-			s.handleTask(i)
+		for {
+			select {
+			case i := <-s.tasks:
```

Comment:
@mlaventure right, but that snipped is just fixing case `c)` in the linked docker issue (https://github.com/moby/moby/issues/31487), `b)` would still be an issue right?